test/spec/proxySpec.js

```diff
@@ -1148,6 +1146,86 @@ describe('proxy', function() {
       // }, 100);
     });
   });
+  describe('Already proxified values', function() {
+    it("shouldn't re-proxify (it should deep clone) proxified values from different instances", function() {
```

Comment:
> If that's the nature of proxies 

Not really, we're voluntarily deep cloning any object that is set to a parent proxified object. 

>  then they will never use proxifiedObject directly, will they?

100% true, or more, our developers will never have two instances of JSONPatcherProxy.